MaterialDesignThemes.Wpf/SnackbarMessageQueueItem.cs

```diff
@@ -47,8 +48,37 @@ internal class SnackbarMessageQueueItem
         public bool IsPromoted { get; }
 
         /// <summary>
-        /// Still display this message even if it is a duplicate.
+        /// Always show this message, even if it's a duplicate
         /// </summary>
-        public bool IgnoreDuplicate { get; }
+        public bool ShowAlways { get; }
+
+        /// <summary>
+        /// Last time this message was shown
+        /// </summary>
+        public DateTime LastShownAt { get; set; }
+
+        public bool MessageExpired()
+        {
+            if (LastShownAt == null)
+                return true;
+            else
+                return LastShownAt <= DateTime.Now.Subtract(Duration);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (!(obj is SnackbarMessageQueueItem message))
+                return false;
+
+            return Content.Equals(message.Content) && ActionContent.Equals(message.ActionContent);
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = -724337062;
+            hashCode = hashCode * -1521134295 + EqualityComparer<object>.Default.GetHashCode(Content);
+            hashCode = hashCode * -1521134295 + EqualityComparer<object>.Default.GetHashCode(ActionContent);
+            return hashCode;
```

Comment:
Though this implementation works, with the new C# value tuple there is a simpler method.
```
return (Content, ActionContent).GetHashCode();
```

Reply:
This requires System.ValueTuple, doesn't it? I remember that being weird with versions of the .NET Framework <= 4.7. In my case I had to download the Nuget-Package for System.ValueTuple.

Reply:
Yes you are correct. In this case I think I am ok taking a dependency on it for the case of `net45`.